### PR TITLE
Feature gate initializers field

### PR DESCRIPTION
**Pull request by liggitt**
The metadata.initializers field should be feature gated and disabled by default while in alpha, especially since enforcement of initializer permission that keeps users from submitting objects with their own initializers specified is done via an admission plugin most clusters do not enable yet.

Not gating the field and tests caused tests added in https://github.com/kubernetes/kubernetes/issues/51429 to fail on clusters that don't enable the admission plugin.

This PR:
* adds an `Initializers` feature gate, auto-enables the feature gate if the admission plugin is enabled
* clears the `metadata.initializers` field of objects on create/update if the feature gate is not set
* marks the e2e tests as feature-dependent (will follow up with PR to test-infra to enable the feature and opt in for GCE e2e tests)

```release-note
Use of the alpha initializers feature now requires enabling the `Initializers` feature gate. This feature gate is auto-enabled if the `Initialzers` admission plugin is enabled.
```

### CODE REVIEW

File: plugin/pkg/admission/initialization/initialization.go [hunk 1]
```diff
@@ -71,6 +73,18 @@ func (i *initializer) Validate() error {
 	if i.config == nil {
 		return fmt.Errorf("the Initializer admission plugin requires a Kubernetes client to be provided")
 	}
+	if i.authorizer == nil {
+		return fmt.Errorf("the Initializer admission plugin requires an authorizer to be provided")
+	}
+
+	if !utilfeature.DefaultFeatureGate.Enabled(features.Initializers) {
+		if err := utilfeature.DefaultFeatureGate.Set(string(features.Initializers) + "=true"); err != nil {
+			glog.Errorf("error enabling Initializers feature as part of admission plugin setup: %v", err)
```
Comment (smarterclayton) [inline plugin/pkg/admission/initialization/initialization.go]: utilruntime.HandleError

**Comment by smarterclayton** [comment]
/approve

Setting feature based on presence of Admission controller is very reasonable - the opposite does not make sense.  Although it may be novel to some people, so we should make sure we communicate it.

**Comment by liggitt** [comment]
cc @kubernetes/sig-api-machinery-api-reviews

File: staging/src/k8s.io/apiserver/pkg/registry/rest/update.go [hunk 2]
```diff
@@ -99,6 +101,12 @@ func BeforeUpdate(strategy RESTUpdateStrategy, ctx genericapirequest.Context, ob
 	}
 	objectMeta.SetGeneration(oldMeta.GetGeneration())
 
+	// Ensure Initializers are not set unless the feature is enabled
+	if !utilfeature.DefaultFeatureGate.Enabled(features.Initializers) {
+		oldMeta.SetInitializers(nil)
```
Comment (caesarxuchao) [inline staging/src/k8s.io/apiserver/pkg/registry/rest/update.go]: Why setting the oldMeta?

Comment (liggitt) [reply staging/src/k8s.io/apiserver/pkg/registry/rest/update.go]: if the feature is disabled, the server should behave as if it doesn't know about the field at all. ideally that would be done in decoding based on field gates, but until then, we mimic that by dropping it from all objects here.

**Comment by caesarxuchao** [comment]
> auto-enables the feature gate if the admission plugin is enabled

Could you add this to the release note? Then lgtm.

**Comment by liggitt** [comment]
> Could you add this to the release note? Then lgtm.

Added

**Pull request merged**